coap/nats/pubsub.go

```diff
@@ -0,0 +1,64 @@
+// Package nats contains NATS-specific message repository implementation.
+package nats
+
+import (
+	"fmt"
+
+	"github.com/sony/gobreaker"
+
+	"github.com/golang/protobuf/proto"
+	"github.com/mainflux/mainflux"
+	broker "github.com/nats-io/go-nats"
+)
+
+var _ mainflux.MessagePublisher = (*natsPublisher)(nil)
+
+const (
+	maxFailedReqs   = 3
+	maxFailureRatio = 0.6
+	prefix          = "channel"
+)
+
+type natsPublisher struct {
+	nc *broker.Conn
+	cb *gobreaker.CircuitBreaker
```

Comment:
While it is perfectly suitable place for a circuit breaker, I would advise that we drop it for now, for the sake of making the code slightly simpler.

Please note that this might be changed in the future.